src/app/tests/yaml/onoff.yaml

```diff
@@ -0,0 +1,54 @@
+# Copyright (c) 2021 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: OnOff Tests
+
+config:
+    cluster: "OnOff"
+    endpoint: 1
+
+tests:
+    - label: "Check on/off attribute value"
+      type: "readAttribute"
+      name: "onOff"
+      response:
+          error: 0
+          value: 0
+          type: "bool"
+
+    - label: "Send On Command"
+      name: "on"
+      response:
+          error: 0
+
+    - label: "Check on/off attribute value"
+      type: "readAttribute"
+      name: "onOff"
+      response:
+          error: 0
+          value: 1
```

Comment:
how do value and type interact? Does the xml not have sufficient information to infer data types? 

All these commands have 'error: 0'. Could I avoid repeating that?

Reply:
> how do value and type interact? Does the xml not have sufficient information to infer data types?

It does. But I have not plugged the real xml definition and the tests generation yet.

> All these commands have 'error: 0'. Could I avoid repeating that?

Definitively. In fact I did built support for that and I forgot to remove those...

Reply:
> how do value and type interact? Does the xml not have sufficient information to infer data types?
> 

I have started to glue together the `value` from the tests and what is stored into the ZAP database in this PR.
I planned  to do a followup where I will add methods to `Test Cluster` that consumes all the supported types. Both for parameters and for response values. This is where I expect most of the `type` mapping plumbing to happens.